pkg/virt-controller/watch/vm/vm.go

```diff
@@ -3191,16 +3191,42 @@ func (c *Controller) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineI
 		return vm, common.NewSyncError(fmt.Errorf(fetchingRunStrategyErrFmt, err), FailedCreateReason), err
 	}
 
-	// Ensure we have ControllerRevisions of any instancetype or preferences referenced by the VM
-	err = c.instancetypeMethods.StoreControllerRevisions(vm)
-	if err != nil {
-		log.Log.Object(vm).Infof("Failed to store Instancetype ControllerRevisions for VirtualMachine: %s/%s", vm.Namespace, vm.Name)
-		c.recorder.Eventf(vm, k8score.EventTypeWarning, common.FailedCreateVirtualMachineReason, "Error encountered while storing Instancetype ControllerRevisions: %v", err)
-		return vm, common.NewSyncError(fmt.Errorf("Error encountered while storing Instancetype ControllerRevisions: %v", err), common.FailedCreateVirtualMachineReason), nil
+	// TODO(lyarwood): Extract all instance type logic into a handler
+	referencePolicy := c.clusterConfig.GetInstancetypeReferencePolicy()
+	switch referencePolicy {
+	case virtv1.Reference:
+		// Ensure we have ControllerRevisions of any instancetype or preferences referenced by the VM
+		if err = c.instancetypeMethods.StoreControllerRevisions(vm); err != nil {
+			log.Log.Object(vm).Infof("failed to store Instancetype ControllerRevisions for VirtualMachine: %s/%s", vm.Namespace, vm.Name)
+			c.recorder.Eventf(vm, k8score.EventTypeWarning, common.FailedCreateVirtualMachineReason, "Error encountered while storing Instancetype ControllerRevisions: %v", err)
+			return vm, common.NewSyncError(fmt.Errorf("Error encountered while storing Instancetype ControllerRevisions: %v", err), common.FailedCreateVirtualMachineReason), nil
+		}
+	case virtv1.Expand, virtv1.ExpandAll:
```

Comment:
Where is the logic specific to `ExpandAll`? Sorry I couldn't find it, probably github hiding it from me. I wanted to make sure we remove the controller revision when `ExpandAll` is used.

Reply:
It's the same block below, the initial conditional to check if we should expand was the only difference. Both `nil` out `spec.{instancetype,preference}` and should be asserted in the unit and functional tests.

Reply:
The actual expand logic is the same, the only difference is the check beforehand to see if we should expand or not. That has been extracted in the following commit per @vladikr 's request and should hopefully be easier to read now https://github.com/kubevirt/kubevirt/pull/13086/commits/57643b2660dc0eb6b5a8417127d50595fca4c44b

Reply:
Ah, got it, thank you. I was grepping for `ExpandAll` :facepalm: